### ISSUE TITLE
TST: Build documentation on Travis-CI?

### ISSUE THREAD

**Issue by mgeier** [jupyter/nbconvert#883]
This would catch errors such as  #878 in the future.

I would make a PR for that, but I don't know how to build the docs only once (for the most recent Python version), since I think it isn't necessary to build it for every Python version.

**Comment by t-makaro** [comment]
Does Travis-CI have the ability to upload build artifacts? If not, I recommend using Circle-CI. This way we can view the rendered docs from each build.

**Comment by MSeal** [comment]
We build this once as a separate task in papermill https://github.com/nteract/papermill/blob/master/tox.ini#L18-L26 via https://github.com/nteract/papermill/blob/master/.travis.yml and it works pretty well. It'd probably be possible to add a push to the docs site with a PR-#### label too, though I haven't worked through those details. I think to upload via travis we'd need to do something like https://docs.travis-ci.com/user/deployment#deploying-to-multiple-providers

**Comment by mgeier** [comment]
I don't think it is necessary to have access to the artifacts. Just checking whether the build succeeds or not would be most helpful IMHO.
If a cell fails to execute, the file name, cell content and traceback should appear in the logs.

If access to the built docs is needed, this can be added later.
